pySDC/projects/DAE/misc/HookClass_DAE.py

```diff
@@ -1,84 +1,86 @@
 from pySDC.core.Hooks import hooks
 
 
-class approx_solution_hook(hooks):
+class error_hook(hooks):
     """
-    Hook class to add the approximate solution to the output generated by the sweeper after each time step
+    Hook class to log the error to the output generated by the sweeper after
+    each time step.
     """
 
     def __init__(self):
-        """
-        Initialization routine for the custom hook
-        """
-        super(approx_solution_hook, self).__init__()
+        """Initialization routine"""
+        super().__init__()
 
     def post_step(self, step, level_number):
-        """
-        Default routine called after each step
-        Args:
-            step: the current step
-            level_number: the current level number
+        r"""
+        Default routine called after each step.
+
+        Parameters
+        ----------
+        step : pySDC.core.Step
+            Current step.
+        level_number : pySDC.core.level
+            Current level number.
         """
 
-        super(approx_solution_hook, self).post_step(step, level_number)
+        super().post_step(step, level_number)
 
         # some abbreviations
         L = step.levels[level_number]
+        P = L.prob
 
         # TODO: is it really necessary to recompute the end point? Hasn't this been done already?
         L.sweep.compute_end_point()
 
+        # compute and save errors
+        # Note that the component from which the error is measured is specified here
+        upde = P.u_exact(step.time + step.dt)
+        err = abs(upde.diff - L.uend.diff)
```

Comment:
But here @pancetta and @brownbaerchen the error is computed over the differential variables. When I would use the generic error hook classes I got the maximum error over both the differential and the algebraic variables. So, I still would like to keep this class at least for the semi-explicit DAEs.  For the fully-implicit DAEs the generic error hook classes does make sense to use!